### PR TITLE
Clear command

### LINKED ISSUE #3840: /clear command
## Description

/clear 

## Justification

i know many servers including my own have this most likely written in our private cores, however i think it should be add here since it is a vanilla mc command

### PR DESCRIPTION
**Pull request by DimBis**
### Relevant issues
* Fixes #3840

## Follow-up

Requires translations:

| Name | Value in eng.ini |
| :--: | :---: |
| `commands.clear.successful` | `Cleared the inventory of {%0}, removing {%1} items` |
| `pocketmine.command.clear.description` | `Clears items from player inventory` |
| `pocketmine.command.clear.usage` | `/clear [player]` |

### CODE REVIEW

File: src/pocketmine/command/defaults/ClearCommand.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				$sender->sendMessage($sender->getServer()->getLanguage()->translateString(TextFormat::RED . "%commands.generic.permission"));
+
+				return true;
+			}
+
+			$player = $sender->getServer()->getPlayer($args[0]);
+
+			if($player instanceof Player){
+
+				$cleared = 0;
```
Comment (lukeeey) [inline src/pocketmine/command/defaults/ClearCommand.php:66]:
```php
$cleared = count($player->getInventory()->getContents();
```
same below

**Review by lukeeey — CHANGES_REQUESTED** [review]

**Comment by lukeeey** [comment]
Can you please at least follow the template or remove the parts that don't apply (there are no API changes here so you can get away with not having that section)

also
image

File: src/pocketmine/command/defaults/ClearCommand.php (hunk 1 above)
Comment (dktapps) [reply src/pocketmine/command/defaults/ClearCommand.php:66]: It's not the same. This loop adds together all of the itemstacks' counts in the inventory (e.g. 1 stack of 64 diamonds cleared will report 64 items).

Comment (lukeeey) [reply src/pocketmine/command/defaults/ClearCommand.php:66]: oh sorry, my bad

File: src/pocketmine/command/defaults/ClearCommand.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+				foreach ($player->getInventory()->getContents() as $content) {
+					$cleared += $content->getCount();
+				}
+
+				$player->getInventory()->clearAll();
+				Command::broadcastCommandMessage($sender, new TranslationContainer("commands.clear.successful", [$player->getName(), $cleared]));
```
Comment (dktapps) [inline src/pocketmine/command/defaults/ClearCommand.php:73]: The code inside this block is repeated.

**Comment by dktapps** [comment]
I can't accept this on stable due to feature freeze. You'll need to target the master branch instead.

**Comment by dktapps** [comment]
This appears to be a near-perfect copy-paste of `KillCommand` with adjustments, explaining why the target selection makes no sense.

**Comment by Ifera** [comment]
Also.. It's only clearing players inventory. Shouldn't this also clear it's armor and cursor inventory as well?

**Comment by DimBis** [comment]
> Also.. It's only clearing players inventory. Shouldn't this also clear it's armor and cursor inventory as well?

I have searched on google for the behavior of the command and i found an article says it ONLY clears the inventory contents and not the armor inventory.

**Comment by DimBis** [comment]
> This appears to be a near-perfect copy-paste of `KillCommand` with adjustments, explaining why the target selection makes no sense. @shoghicp is cursing us from beyond the grave.
Well nearly all of the built in commands are the same exact shit so i didn't bother myself to unshit it.

**Comment by dktapps** [comment]
> Well nearly all of the built in commands are the same exact shit so i didn't bother myself to unshit it.

Bad code isn't an excuse to write more bad code. I won't merge something that I have to rewrite.

**Comment by DimBis** [comment]
> > Well nearly all of the built in commands are the same exact shit so i didn't bother myself to unshit it.
> 
> Bad code isn't an excuse to write more bad code. I won't merge something that I have to rewrite.

k i don't remember asking tho

**Comment by DimBis** [comment]
Also if you don't want a bad code to rewrite why did you decide to maintain pocketmine in the first place?!?!?!

**Pull request closed without merging**